fontforge/encoding.c

```diff
@@ -1210,29 +1211,41 @@ return( NULL );
     continue;
 	    val = strtol(pt,&end,10);
 	    while ( isspace(*end)) ++end;
-	    if ( strncmp(end,bcsr,strlen(bcsr))==0 )
+	    in_is_single = 0;
+	    if ( strncmp(end,bcsr,strlen(bcsr))==0 ) {
 		in = cmt_coderange;
-	    else if ( strncmp(end,bndr,strlen(bndr))==0 )
+	    } else if ( strncmp(end,bndr,strlen(bndr))==0 ) {
 		in = cmt_notdefs;
-	    else if ( strncmp(end,bcr,strlen(bcr))==0 )
+	    } else if ( strncmp(end,bcr,strlen(bcr))==0 ) {
 		in = cmt_cid;
+	    } else if ( strncmp(end,bcc,strlen(bcc))==0 ) {
+		in = cmt_cid;
+		in_is_single = 1;
+	    }
 	    if ( in!=cmt_out ) {
 		pos = cmap->groups[in].n;
 		cmap->groups[in].ranges = ExtendArray(cmap->groups[in].ranges,&cmap->groups[in].n,val);
 	    }
 	} else if ( strncmp(pt,"end",3)== 0 )
 	    in = cmt_out;
 	else {
+	    // Read the first bracketed code.
 	    if ( *pt!='<' )
 	continue;
 	    cmap->groups[in].ranges[pos].first = strtoul(pt+1,&end,16);
 	    if ( *end=='>' ) ++end;
 	    while ( isspace(*end)) ++end;
-	    if ( *end=='<' ) ++end;
-	    cmap->groups[in].ranges[pos].last = strtoul(end,&end,16);
+	    if (in_is_single) {
+	      cmap->groups[in].ranges[pos].last = cmap->groups[in].ranges[pos].first;
+	    } else {
+	      // Read the second bracketed code.
+	      if ( *end=='<' ) ++end;
+	      cmap->groups[in].ranges[pos].last = strtoul(end,&end,16);
+	      if ( *end=='>' ) ++end;
+	    }
 	    if ( in!=cmt_coderange ) {
-		if ( *end=='>' ) ++end;
```

Comment:
I moved this up because it seemed appropriate to parse the entire bracketed item regardless of whether something else follows.